lib/IDE/SourceEntityWalker.cpp

```diff
@@ -645,9 +652,33 @@ ASTWalker::PreWalkAction SemaAnnotator::walkToTypeReprPre(TypeRepr *T) {
         auto Continue = passReference(ModD, {ident, IdT->getLoc()});
         return Action::StopIf(!Continue);
       }
-      auto Continue = passReference(
-          VD, Type(), IdT->getNameLoc(),
-          ReferenceMetaData(SemaReferenceKind::TypeRef, llvm::None));
+
+      Type Ty = T->getType();
```

Comment:
It's a bit strange to walk TypeReprs here and then expect them to have been resolved already with a type set, for several reasons. The main one being that it creates an order dependency on primary file checking having been run. Ideally IDE tools would rely completely on lazy request-driven type checking, but that means calling accessors on Decls and Exprs that return Types instead of walking TypeReprs. It also just feels like a layering violation to have a syntactic construct point at a semantic construct.